ChopChop/ViewModels/RecipeFormViewModel.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  RecipeFormViewModel.swift
+//  ChopChop
+//
+//  Created by Cao Wenjie on 20/3/21.
+//
+import SwiftUI
+import Combine
+import GRDB
+
+class RecipeFormViewModel: ObservableObject {
+
+    private var existingRecipe: Recipe?
+    @Published var hasError: Bool = false
+    private var recipeId: Int64?
+    private(set) var errorMessage = ""
+    private(set) var isEdit = false
+    private let storageManager = StorageManager()
+    private var recipeCategoryCancellable = Set<AnyCancellable>()
+    @Published var recipeName = ""
+    @Published var serving = "" {
+        didSet {
+            ensureValidServing()
+        }
+    }
+    @Published var allRecipeCategories = [RecipeCategory]()
+    @Published var recipeCategory = ""
+    @Published var difficulty: Difficulty = .veryEasy
```

Comment:
make it optional, i made a DifficultyView just use later

Reply:
I think I need this to not be optional cos this is used to render the options in the menu picker.